WordPress/gutenberg
# Mobile - Global styles: Block-specific styles

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/2195

This PR is a continuation of https://github.com/WordPress/gutenberg/pull/30684 and https://github.com/WordPress/gutenberg/pull/30810

## Description
This PR adds support to render the theme's block-specific styles. It also adds a reset option in the color picker.

## How has this been tested?

<details><summary>Precondition</summary>
<hr>
To be able to test this feature you need a WordPress 5.8 site running with the Gutenberg plugin and a FSE theme.

- Create a JN testing site with the following options:
  - **Don't** include Jetpack.
  - Include Gutenberg plugin.
  - Include FSE-enabled TT1 Blocks Theme.
- Once the site is created activate the TT1 Blocks Theme.
- Install the WordPress Beta Tester Plugin.
- Activate WordPress Beta Tester.
- Go to Tools > Beta testing.
- Enable the Bleeding edge option.
- Go to Dashboard > Updates.
- Click on Update to the latest nightly.

Now you should have all needed to test this PR.
</p>
</details>

### In the WP Admin dashboard:
- Go to the Site editor.
- To the up right of the page, tap on Global styles
   - <img width="125" alt="Screenshot 2021-06-24 at 12 27 25" src="https://user-images.githubusercontent.com/4885740/123247893-9b1f8480-d4e7-11eb-8876-ae6e83e11ecd.png">
- Click on By block type
- Change the default styles (Background color and text color) of different blocks like: `Group`, `Paragraph`, `Heading`, `List`, `Verse`, `Media&Text`, `Columns`, `Column`, `Buttons`.

### On the mobile editor

- Open the WordPress app with metro running from these PRs:
   - `Android` https://github.com/wordpress-mobile/WordPress-Android/pull/14597
   - `iOS` https://github.com/wordpress-mobile/WordPress-iOS/pull/16411
- Log in to your JN site (self-hosted option). 
- Create a new post
- Add the blocks you changed the styles to
- **Expect** to see the new styles in the mobile editor.

### Reset custom color selection

- Add a Paragraph block
- Set a text color
- Expect to see the reset option available
- Tap on reset
- **Expect** to see the default color and not the custom one

## Screenshots
iOS|Android|Reset colors
-|-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/123261461-19cfee00-d4f7-11eb-9e4e-432f41f77c5a.png" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/123261477-1c324800-d4f7-11eb-80cb-f76cad28b129.png" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/123262237-f2c5ec00-d4f7-11eb-875e-1ae54a06e92d.gif" width="200" /></kbd>

## Types of changes
New feature

## Checklist:
